NumLib/Function/Interpolation.h

```diff
@@ -16,38 +16,69 @@
 namespace NumLib
 {
 
-/**
+namespace detail
+{
+
+template<unsigned DOFOffset, typename NodalValues, typename ShapeMatrix>
+void shapeFunctionInterpolate(
+        const NodalValues &/*nodal_values*/,
+        const ShapeMatrix &/*shape_matrix_N*/)
+{}
+
+template<unsigned DOFOffset, typename NodalValues, typename ShapeMatrix, typename... Numbers>
```

Comment:
This is in detail namespace. The documentation is given below for the function that is to be used by users. The current implementation restricts the type to `double`, as expressed by `Numbers`, i.e., you should not pass a vector. Furthermore, the name of the argument is descriptive, at least I hope so.